lazy-imports-import.html

```diff
@@ -22,23 +23,26 @@
       var imports = fromElement.querySelectorAll(query);
       var promises = [];
       for (var i = 0; i < imports.length; i++) {
-        promises.push(new Promise(function(resolve) {
-          var href = imports[i].getAttribute('href');
-          var onLoad = function() {
-            resolve({loaded: href});
-          };
-          var onFailure = function() {
-            resolve({failed: href});
-          };
-          var resolved = callingElement.resolveUrl(href);
-          if (!__loadedImports[resolved]) {
-            __loadedImports[resolved] = true;
+        var href = imports[i].getAttribute('href');
+        var resolvedUrl = callingElement.resolveUrl(href);
+
+        if (__failedImports[resolvedUrl] || !__loadedImports[resolvedUrl]) {
```

Comment:
@justinfagnani Just wanted to see if you had any thoughts on my reply that I will need to address.